Realm/RLMRealm.mm

```diff
@@ -628,23 +628,20 @@ - (void)deleteObject:(RLMObject *)object {
     RLMDeleteObjectFromRealm(object, self);
 }
 
-- (void)deleteObjects:(id)array {
+- (void)deleteObjects:(NSObject<NSFastEnumeration> *)array {
```

Comment:
Previously, this method took an argument of type `id`, which is not very type safe. As such, we threw an exception when it was neither a type that conformed to `NSFastEnumeration` or a Realm collection. We'd like to mark this argument as `id<NSFastEnumeration>`, as the arguments of the other methods were, to ensure that only enumerable collections are passed in. Since Realm collections are themselves enumerable, this is a reasonable API.

In order to check whether or not an object is a Realm collection, we have to call `respondsToSelector:`, which is defined on the `NSObject` protocol. I think it's fair to assume that our API will only be used with NSObject subclasses (Swift collections should instead use a `SequenceType` variant of this method), so I marked its argument `NSObject<NSFastEnumeration>` and I did the same to the other methods for consistency.

Let me know if you think this is reasonable.